src/generate/CodeModelAzImpl.ts

```diff
@@ -3433,4 +3557,311 @@ export class CodeModelCliImpl implements CodeModelAz {
+                    ],
+                    converter,
+                ),
+            ],
+            ['azExample', new RenderInput(['commandStringItems'], {})],
+        ]);
+
+        
+        This means we need extension => properties: name, sort by name
+                           commandGroup => properties: name, cliKey, sort by name
+                           command => properties: name,
+                           method => properties: nameAz, cliKey, sort by nameAz
+                           methodParameter => properties: mapsTo, type, description, cliKey, namePython, and filter those parameters that 
+                                isFlattened == true and methodParameter_type != constants, isPolyOfSimple != true, isDiscrminator != true
+
+                                and converter
+
+        data = model.getModelData('extension', inputProperties, dependencies);
+
+     */
+
+    public getModelData(
+        layer: CodeModelTypes,
+        inputProperties: Map<CodeModelTypes, RenderInput>,
+        dependencies: DataGraph,
+    ): unknown | any[] {
+        if (
+            isNullOrUndefined(layer) ||
+            isNullOrUndefined(dependencies) ||
+            dependencies.length < 0
+        ) {
+            return {};
+        }
+        const type: CodeModelTypes = layer;
+        const Type = Capitalize(type);
+        const renderInput: RenderInput = inputProperties.get(type);
+        const sortBy = renderInput.sortBy;
+        const props = renderInput.properties;
+        const conditions = renderInput.conditions;
+        const converter = renderInput.converter;
+        const data = {};
+        data['has' + Type] = false;
+        data[Type + 's'] = [];
+        const items = [];
+        if (this['SelectFirst' + Type]()) {
+            data['has' + Type] = true;
+            do {
+                let item = {};
+                let hasFiltered = false;
+                if (!isNullOrUndefined(props) && Array.isArray(props) && props.length > 0) {
+                    for (const prop of props) {
+                        item[prop] = this[Type + '_' + Capitalize(prop)];
+                        if (!isNullOrUndefined(converter.get(prop))) {
+                            item[prop] = converter.get(prop)(item[prop]);
+                        }
+                    }
+                    for (const condition of conditions) {
+                        if (this[Type + '_' + Capitalize(condition[0])] === condition[1]) {
+                            hasFiltered = true;
+                            break;
+                        }
+                    }
+                }
+                if (hasFiltered) {
+                    continue;
+                }
+
+                if (dependencies.length > 0) {
+                    // const d = dependencies.shift();
+                    for (let i = 0; i < dependencies.length; i++) {
+                        if (layer === dependencies[i][0]) {
```

Comment:
Is it possible to use map functions here? E.g.  map.has(), map.get() etc.

Reply:
changed it into filter and forEach